src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -249,8 +249,8 @@ def get_session_id(self) -> int:
 
     @_Decorator.wrap_exception
     def _get_current_parameter(self, param: str, quoted: bool = True) -> Optional[str]:
-        name = getattr(self._conn, param) or self._get_string_datum(
-            f"SELECT CURRENT_{param.upper()}()"
+        name = self._get_string_datum(f"SELECT CURRENT_{param.upper()}()") or getattr(
+            self._conn, param
```

Comment:
In what case would `self._get_string_datum` return something falsey?

Reply:
Some edge cases like if the role does not have priviledge to run `select current_role()`